### ISSUE TITLE
Validation data reader copies preprocessing

### ISSUE THREAD

**Issue by ndryden** [LLNL/lbann#1019]
When creating a validation data reader from a percent of the training data, the created data reader is made by copying the training data reader. This also copies the associated preprocessing setup.

This is very likely not what we want, since it will typically apply the training data reader's random augmentation to the validation data, making the validation accuracy not representative of the test time accuracy.
